### PR TITLE
scripts for performance tracking system

### PR DESCRIPTION
**Pull request by jzuniga-amd**
This PR adds new scripts to generate internal performance reports to track the optimization efforts (see the README file for more details).

### CODE REVIEW

**Comment by amd-jnovotny** [comment]
I noticed that we have a docs section on benchmarking. Do you think this information needs to be added to docs too? Or is it something that only developers/contributors would be concerned with?

**Review by AGonzales-amd — APPROVED** [review]

**Comment by jzuniga-amd** [comment]
> I noticed that we have a docs section on benchmarking. Do you think this information needs to be added to docs too? Or is it something that only developers/contributors would be concerned with?

Correct, this is intended only for internal use (by developers and CI)

**Comment by samjwu** [comment]
%PTS
geqrf s small
syevd s small
syevdx s small

File: scripts/perf/rocsolver-perfoptim-suite.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+    size = sizenormal
+    for s in size:
+        row = {'name': precision+suite, 'name_test': suite, 'function': fn, 'precision': precision, 'n': s}
+        yield (row, s, f'-f {fn} -r {precision} -n {s} {common}')
+
+"""
+POTRFBATCH tests are run with the given precision and sizes
+"""
+def potrfBatch_suite(*, suite, precision, sizenormal, sizebatch):
+    fn = 'potrf_batched'
```
Comment (tfalders) [inline scripts/perf/rocsolver-perfoptim-suite.py:154]: Other tests test the strided_batched function. Is there a reason POTRF is testing the batched function?

File: scripts/perf/rocsolver-perfoptim-suite.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+    process = Popen(cmd, stdout=PIPE, stderr=PIPE)
+    vprint('executing {}'.format(' '.join(cmd)))
+    stdout, stderr = process.communicate()
+    return (str(stdout, encoding='utf-8', errors='surrogateescape'),
+            str(stderr, encoding='utf-8', errors='surrogateescape'),
+            process.returncode)
+
+"""
+EXECUTE_BENCHMARKS collects the arguments for the benchmark client, calls
+the client, gets the resulting time, and put everything in file of screen
```
Comment (tfalders) [inline scripts/perf/rocsolver-perfoptim-suite.py]: Spelling on or?

**Review by tfalders — APPROVED** [review]

**Review by jmachado-amd — APPROVED** [review]
The code reads well, and everything looks good to me. Yet, I am trusting CI to validate this one, as I have not tried the scripts myself.

File: scripts/perf/perfoptim-suite [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+# size cases to run:
+# valid options are: (default is large)
+# small  -> see definitions in rocsolver-perfoptim-suite.py for included size values  
+# medium -> see definitions in rocsolver-perfoptim-suite.py for included size values
+# large  -> see definitions in rocsolver-perfoptim-suite.py for included size values
+# (note: select only one as small is a sub-set of medium which is a sub-set of large)
+Clist="small medium large"
+ 
+
+# Get an validate input arguments:
```
Comment (jmachado-amd) [inline scripts/perf/perfoptim-suite]: Typo? I think you mean "Get and validate input...".

File: scripts/perf/rocsolver-perfoptim-suite.py (hunk 1 above)
Comment (jzuniga-amd) [reply scripts/perf/rocsolver-perfoptim-suite.py:154]: Yeah, all batch functions in cusolver are strided_batched, but for some reason, potrf is _batched

File: scripts/perf/rocsolver-perfoptim-suite.py (hunk 2 above)
Comment (jzuniga-amd) [reply scripts/perf/rocsolver-perfoptim-suite.py]: corrected

File: scripts/perf/perfoptim-suite (hunk 3 above)
Comment (jzuniga-amd) [reply scripts/perf/perfoptim-suite]: corrected

**Pull request merged**